reddit_decider/__init__.py

```diff
@@ -555,6 +559,178 @@ def get_variant_for_identifier_without_expose(
 
         return variant
 
+    def get_all_variants_without_expose(self) -> Dict[str, Optional[str]]:
+        """Return a dict of experiment name strings as keys and
+            variant names as the values (if a variant is assigned
+            or None otherwise). All available experiments get bucketed.
+            Exposure events are not emitted.
+
+        The `expose()` function is available to be manually called afterward to emit
+        exposure event.
+
+        However, experiments in Holdout Groups will still send an exposure for
+        the holdout parent experiment, since it is not possible to
+        manually expose the holdout later (because after exiting this function,
+        it's impossible to know if a returned `None` or `"control_1"` string
+        came from the holdout group or its child experiment).
+
+        :return: dict of experiment name strings as keys and
+            variant names (or `None`) as the values
+        """
+        decider = self._get_decider()
+        if decider is None:
+            logger.error("Encountered error in _get_decider()")
+            return None
+
+        context_fields = self._decider_context.to_dict()
+        ctx = rust_decider.make_ctx(context_fields)
+        ctx_err = ctx.err()
+        if ctx_err is not None:
+            logger.info(f"Encountered error in rust_decider.make_ctx(): {ctx_err}")
+            return {}
```

Comment:
@julieahnt I return empty dict on error here, is that ok?

Reply:
If the GQL does not need to call get_all_experiment_names , SDK will need provide the same data from existing bulk query:
https://gql.reddit.com/?variablesGz=eJyr5lJQUCpITE8NqSxIVbICsovy0zJzUpV0wBIQjl9iLlgut7i4ICexOCMjsUSJqxYAHxUS0Q%3D%3D

Reply:
with the dict in this format:
```
{
        "id": "700",
        "name": "variant_1",
        "version": "56",
        "experimentName": "xxx"
}
```
?

Reply:
GQL actually calls each experiment object to construct these info. 
https://github.snooguts.net/reddit/reddit-service-graphql/blob/master/graphql-py/graphql_api/models/experiment.py#L257-L275

Reply:
since gql doesn't include experiments that return variant of `None`, I've removed those experiments from the return dict.
Will update the format of the response into an array of dicts with this format:
```
{
        "id": "700",
        "name": "variant_1",
        "version": "56",
        "experimentName": "xxx"
}
```
in a follow-up pr.